src/execution/operator/join/physical_piecewise_merge_join.cpp

```diff
@@ -190,7 +346,174 @@ class PiecewiseMergeJoinState : public OperatorState {
+		data_pointers[i] = row_ptr;
+		row_ptr += row_width;
+	}
+	// Unswizzle the offsets back to pointers (if needed)
+	if (!sorted_data.layout.AllConstant() && info.state.external) {
+		RowOperations::UnswizzlePointers(sorted_data.layout, data_ptr, read_state.payload_heap_handle->Ptr(), next);
+	}
+
+	// Deserialize the payload data
+	for (idx_t col_idx = 0; col_idx < sorted_data.layout.ColumnCount(); col_idx++) {
+		const auto col_offset = sorted_data.layout.GetOffsets()[col_idx];
+		RowOperations::Gather(addresses, info.result, payload.data[left_cols + col_idx],
+		                      *FlatVector::IncrementalSelectionVector(), result_count, col_offset, col_idx);
+	}
+}
+
+static void MergeJoinPinSortingBlock(SBScanState &scan, const idx_t block_idx) {
+	scan.SetIndices(block_idx, 0);
+	scan.PinRadix(block_idx);
+
+	auto &sd = *scan.sb->blob_sorting_data;
+	if (block_idx < sd.data_blocks.size()) {
+		scan.PinData(sd);
+	}
+}
+
+static data_ptr_t MergeJoinRadixPtr(SBScanState &scan, const idx_t entry_idx) {
+	scan.entry_idx = entry_idx;
+	return scan.RadixPtr();
+}
+
+static idx_t MergeJoinSimpleBlocks(PiecewiseMergeJoinState &lstate, MergeJoinGlobalState &rstate, bool *found_match,
+                                   const ExpressionType comparison) {
+	const auto cmp = MergeJoinComparisonValue(comparison);
+
+	// The sort parameters should all be the same
+	auto &lsort = *lstate.lhs_global_state;
+	auto &rsort = rstate.rhs_global_sort_state;
+	D_ASSERT(lsort.sort_layout.all_constant == rsort.sort_layout.all_constant);
+	const auto all_constant = lsort.sort_layout.all_constant;
+	D_ASSERT(lsort.external == rsort.external);
+	const auto external = lsort.external;
+
+	// There should only be one sorted block if they have been sorted
+	D_ASSERT(lsort.sorted_blocks.size() == 1);
+	SBScanState lread(lsort.buffer_manager, lsort);
+	lread.sb = lsort.sorted_blocks[0].get();
+
+	const idx_t l_block_idx = 0;
+	idx_t l_entry_idx = 0;
+	const auto lhs_not_null = lstate.lhs_count - lstate.lhs_has_null;
+	MergeJoinPinSortingBlock(lread, l_block_idx);
+	auto l_ptr = MergeJoinRadixPtr(lread, l_entry_idx);
+
+	D_ASSERT(rsort.sorted_blocks.size() == 1);
+	SBScanState rread(rsort.buffer_manager, rsort);
+	rread.sb = rsort.sorted_blocks[0].get();
+
+	const auto cmp_size = lsort.sort_layout.comparison_size;
+	const auto entry_size = lsort.sort_layout.entry_size;
+
+	idx_t right_base = 0;
+	for (idx_t r_block_idx = 0; r_block_idx < rread.sb->radix_sorting_data.size(); r_block_idx++) {
+		// we only care about the BIGGEST value in each of the RHS data blocks
+		// because we want to figure out if the LHS values are less than [or equal] to ANY value
+		// get the biggest value from the RHS chunk
+		MergeJoinPinSortingBlock(rread, r_block_idx);
+
+		auto &rblock = rread.sb->radix_sorting_data[r_block_idx];
+		const auto r_not_null = SortedBlockNotNull(right_base, rblock.count, rstate.rhs_count - rstate.rhs_has_null);
+		if (0 == r_not_null) {
```

Comment:
Somehow this order offends me :) Total nit but could we change this to `r_not_null == 0`?

Reply:
Yeah sorry this is an old habit called ['Yoda conditions'](https://en.wikipedia.org/wiki/Yoda_conditions) from before widespread use of lint to avoid accidental assignment in tests. I'll switch it.